CRM/Core/BAO/CustomField.php

```diff
@@ -164,6 +164,41 @@ public static function create($params) {
     return $customField;
   }
 
+  /**
+   * Create/ update several fields at once in a mysql efficient way.
+   *
+   * https://lab.civicrm.org/dev/core/issues/1093
+   *
+   * The intention is that apiv4 would expose any BAO with bulkSave as a new action.
+   *
+   * @param array $bulkParams
+   *   Array of arrays as would be passed into create
+   * @param array $defaults
+   *  Default parameters to be be merged into each of the params.
+   */
+  public static function bulkSave($bulkParams, $defaults) {
+    $sql = [];
+    $customFields = [];
+    foreach ($bulkParams as $fieldParams) {
+      $params = array_merge($defaults, $fieldParams);
+      $operation = empty($params['id']) ? 'add' : 'modify';
+      $customField = self::createCustomFieldRecord($params);
+      // Only doing 'add' for now - so hard coded & no indexExists.
```

Comment:
What would it take to get this working with update? Then it would be ready for api4.

Reply:
oh - I think I mostly altered it but not in this spot - will fix